### ISSUE TITLE
mariadb produces .pc file with borked includedir=

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#35143]
## Issue description

The .pc / pkg-config file from mariadb has wrong includedir= directive:

```
$ grep includedir= $(nix-build -A mariadb)/share/mysql/pkgconfig/mariadb.pc
includedir=${prefix}//nix/store/m55f5pgid0rhvixg16gi1q3yfbdif2kr-mariadb-10.2.12/include/mysql
```

The literal part after "${prefix}" is correct, but _with_ "${prefix}" it most definitely is wrong.

This problem is currently blocking upgrade of ntopng.

## Technical details

[Issue apparent in nixpkgs since a looong time ago, `nix-info` is irrelevant. I wonder why nobody has noticed.]

**Comment by bjornfor** [comment]
Also, shouldn't .pc files be in ```$out/lib/pkgconfig```?

**Comment by emmanuelrosa** [comment]
What's the content of the entire `$(nix-build -A mariadb)/share/mysql/pkgconfig/mariadb.pc` file? Perhaps there's a line in the file which looks like `prefix=/nix/store/...`

I ask because the issue would be with `pkgconfig` not `mariadb`, which would affect a LOT of packages.

**Comment by bjornfor** [comment]
(Running ```cat $(nix-build -A mariadb)/share/mysql/pkgconfig/mariadb.pc``` is so easy, I see the point in pasting that file here.)

Yes, there is ```prefix=/nix/store/...``` in there, as it should be. Nothing wrong with pkgconfig, it's the Nix mariadb package file that's causing trouble. The problem is that once ${prefix} gets expanded, you end up with a ```/nix/store/...-mariadb../nix/store/...``` path (double).

**Comment by emmanuelrosa** [comment]
Try building from this: https://github.com/emmanuelrosa/nixpkgs/commit/04d698fad462af0008cc570c02f5e81d8dc10942

**Comment by bjornfor** [comment]
I started a build last night with this patch: https://github.com/bjornfor/nixpkgs/commit/d5299daa9f496fb5f4e41ece6d14d3c467cdcfd8.

I think both patches fixes the .pc file.

**Comment by bjornfor** [comment]
But due to the multiple-outputs stuff going on in the mariadb expression, I need more time to verify the patch.

**Comment by bjornfor** [comment]
Fixed by #40007.
